tests/module/ebuild/test_repo_objs.py

```diff
@@ -109,6 +109,198 @@ def test_longdesc(self):
+    <email>nomembers@example.com</email>
+    <name>Project with no members</name>
+    <url>https://projects.example.com/nomembers</url>
+    <description>Here is the description</description>
+  </project>
+  <project>
+    <email>nolead@example.com</email>
+    <name>Project with no lead</name>
+    <url>https://projects.example.com/nolead</url>
+    <description>Here is the description</description>
+    <member>
+      <email>fulldev1@example.com</email>
+      <name>Full Dev</name>
+      <role>Somebody</role>
+    </member>
+    <member>
+      <email>dev1@example.com</email>
+    </member>
+  </project>
+  <project>
+    <email>lead@example.com</email>
+    <name>Project with a lead</name>
+    <url>https://projects.example.com/lead</url>
+    <description>Here is the description</description>
+    <member>
+      <email>fulldev2@example.com</email>
+      <name>Full Dev</name>
+      <role>Somebody</role>
+    </member>
+    <member is-lead="1">
+      <email>dev2@example.com</email>
+    </member>
+  </project>
+  <project>
+    <email>subprojects@example.com</email>
+    <name>Project with non-recursive subprojects</name>
+    <url>https://projects.example.com/subprojects</url>
+    <description>Here is the description</description>
+    <member>
+      <email>fulldev3@example.com</email>
+      <name>Full Dev</name>
+      <role>Somebody</role>
+    </member>
+    <member is-lead="1">
+      <email>dev3@example.com</email>
+    </member>
+    <subproject ref="nolead@example.com"/>
+    <subproject ref="lead@example.com"/>
+  </project>
+  <project>
+    <email>recursive-subprojects@example.com</email>
+    <name>Project with recursive subprojects</name>
+    <url>https://projects.example.com/recursive-subprojects</url>
+    <description>Here is the description</description>
+    <member>
+      <email>fulldev4@example.com</email>
+      <name>Full Dev</name>
+      <role>Somebody</role>
+    </member>
+    <member>
+      <!-- this one is common -->
+      <email>dev1@example.com</email>
+    </member>
+    <member is-lead="1">
+      <email>dev4@example.com</email>
+    </member>
+    <subproject ref="nolead@example.com" inherit-members="1"/>
+    <subproject ref="lead@example.com"/>
+  </project>
+  <project>
+    <email>super-recursive-subprojects@example.com</email>
+    <name>Project with double recursion</name>
+    <url>https://projects.example.com/super-recursive-subprojects</url>
+    <description>Here is the description</description>
+    <member>
+      <email>fulldev5@example.com</email>
+      <name>Full Dev</name>
+      <role>Somebody</role>
+    </member>
+    <member is-lead="1">
+      <!-- this one is common -->
+      <email>dev2@example.com</email>
+    </member>
+    <member>
+      <email>dev5@example.com</email>
+    </member>
+    <subproject ref="recursive-subprojects@example.com" inherit-members="1"/>
+  </project>
+</projects>"""
+        return repo_objs.ProjectsXml(data_source(s.encode('utf-8')))
+
```

Comment:
Probably need a basic test for the case where projects.xml doesn't exist or is empty. I'd make get_projects_xml() accept a string param defaulting to None in which case it gets the string already defined there. Then have a test pass it the empty string and check viability, etc.

Reply:
I don't think we should accept empty `projects.xml`. Non-existing sure but empty is invalid XML, isn't it?

Reply:
Sure, but I'd still make sure the parsing mechanism outputs an error/warning for that case and ignores it instead of throwing an exception and outputting a backtrace. I've been trying to make pkgcore more acceptable of bad input and logging/ignoring it so then pkgcheck can catch/flag those cases.

Reply:
Ok, made it lenient with `logger.error()`.